spec/services/ingest_iati_activities_spec.rb

```diff
@@ -6,366 +6,387 @@
+          uksa = create(:organisation, name: "UKSA", iati_reference: "GB-GOV-EA31")
+          _existing_project = create(:project_activity,
+            previous_identifier: "GB-GOV-13-GCRF-UKSA_TZ_UKSA-021",
+            organisation: uksa,
+            ingested: true)
 
-        legacy_activities = File.read("#{Rails.root}/spec/fixtures/activities/uksa/with_transactions.xml")
+          legacy_activities = File.read("#{Rails.root}/spec/fixtures/activities/uksa/with_transactions.xml")
 
-        described_class.new(delivery_partner: uksa, file_io: legacy_activities).call
+          described_class.new(delivery_partner: uksa, file_io: legacy_activities).call
 
-        expect_any_instance_of(Activity).not_to receive(:save!)
-        expect_any_instance_of(Transaction).not_to receive(:save!)
-        expect_any_instance_of(Budget).not_to receive(:save!)
-        expect_any_instance_of(PlannedDisbursement).not_to receive(:save!)
+          expect_any_instance_of(Activity).not_to receive(:save!)
+          expect_any_instance_of(Transaction).not_to receive(:save!)
+          expect_any_instance_of(Budget).not_to receive(:save!)
+          expect_any_instance_of(PlannedDisbursement).not_to receive(:save!)
+        end
       end
-    end
 
-    context "when an activity is invalid" do
-      it "raises an error loudly so the team are aware a record didn't save and can review the data" do
-        uksa = create(:organisation, name: "UKSA", iati_reference: "GB-GOV-EA31")
-        legacy_activities = File.read("#{Rails.root}/spec/fixtures/activities/uksa/invalid_activity.xml")
+      context "when an activity is invalid" do
+        it "raises an error loudly so the team are aware a record didn't save and can review the data" do
+          uksa = create(:organisation, name: "UKSA", iati_reference: "GB-GOV-EA31")
+          legacy_activities = File.read("#{Rails.root}/spec/fixtures/activities/uksa/invalid_activity.xml")
+
+          service = described_class.new(delivery_partner: uksa, file_io: legacy_activities)
 
-        service = described_class.new(delivery_partner: uksa, file_io: legacy_activities)
+          expect { service.call }.to raise_error(ActiveRecord::RecordInvalid)
+        end
+      end
+    end
+  end
 
-        expect { service.call }.to raise_error(ActiveRecord::RecordInvalid)
+  context "AMS activities" do
+    context "Newton fund activities" do
+      it "handles activities without an `actual_start_date` (`activity-date` type 2)" do
+        ams = create(:organisation, name: "Academy of Medical Sciences", iati_reference: "GB-COH-03520281")
+        legacy_activities = File.read("#{Rails.root}/spec/fixtures/activities/ams/newt/fake_activity_with_dates.xml")
+        programme = create(:programme_activity, organisation: ams)
+        # Temporary fix until we have the actual programme-project mappings
+        allow_any_instance_of(LegacyActivity).to receive(:find_parent_programme).and_return(programme)
```

Comment:
Nice stub! I think we should have stubbed the inital use of the mappings file in the same way for UKSA to decouple the 2. I reckon we'll have to use this more rather than add files and use it less.